src/normaliz.cc

```
@@ -617,21 +686,73 @@ static Obj _NmzConePropertyImpl(Obj cone, Obj prop)
+        return NmzHilbertQuasiPolynomialToGAP(C->getHilbertSeries());
+    
+    case libnormaliz::ConeProperty::WeightedEhrhartQuasiPolynomial:
+        return NmzWeightedEhrhartQuasiPolynomialToGAP(C->getIntData());
+    
+    case libnormaliz::ConeProperty::IsTriangulationNested:
+        return C->isTriangulationNested() ? True : False;
+        
+    case libnormaliz::ConeProperty::IsTriangulationPartial:
+        return C->isTriangulationPartial() ? True : False;
+    
+    case libnormaliz::ConeProperty::ConeDecomposition:
+        return NmzBoolMatrixToGAP(C->getOpenFacets());
+    
+    case libnormaliz::ConeProperty::ExternalIndex:
+        return NmzIntToGAP(C->getSublattice().getExternalIndex());
+    
+    case libnormaliz::ConeProperty::InternalIndex:
+        return NmzIntToGAP(C->getIndex());
+    
+    case libnormaliz::ConeProperty::WitnessNotIntegrallyClosed:
+        return NmzVectorToGAP(C->getWitnessNotIntegrallyClosed());
+    
+    case libnormaliz::ConeProperty::UnitGroupIndex:
+        return NmzIntToGAP(C->getUnitGroupIndex()); 
 
 //  the following properties are compute options and do not return anything
-    case libnormaliz::ConeProperty::DualMode:
     case libnormaliz::ConeProperty::DefaultMode:
     case libnormaliz::ConeProperty::Approximate:
     case libnormaliz::ConeProperty::BottomDecomposition:
     case libnormaliz::ConeProperty::KeepOrder:
-        return True;    // FIXME: appropriate value?
+    case libnormaliz::ConeProperty::NoBottomDec:
+    case libnormaliz::ConeProperty::PrimalMode:
+    case libnormaliz::ConeProperty::Symmetrize:
+    case libnormaliz::ConeProperty::NoSymmetrization:
+    case libnormaliz::ConeProperty::BigInt:
+    case libnormaliz::ConeProperty::NoNestedTri:
+    case libnormaliz::ConeProperty::HSOP:
+        ErrorQuit("Cone property is input only", 0, 0);
 
     default:
-        // Case not handled. Should signal an error
-        break;
+        ErrorQuit("Unknown cone property", 0, 0);
```

Calling `ErrorQuit` is wrong! It'll `longjmp` out, bypassing the C++ `catch`, preventing any C++ locals from deconstructing etc.

As an alternative, you could throw an special exception, and modify FUNC_END to trigger ErrorQuit for that. Or just print the error message using `Pr` and return fail. Or come up with some other mechanism.